src/main/kotlin/it/pagopa/wallet/cdc/PaymentWalletsLogEventsStream.kt

```diff
@@ -36,33 +41,48 @@ class PaymentWalletsLogEventsStream(
 
     fun streamPaymentWalletsLogEvents(): Flux<Document> {
         val flux: Flux<Document> =
-            reactiveMongoTemplate
-                .changeStream(
-                    changeStreamOptionsConfig.collection,
-                    ChangeStreamOptions.builder()
-                        .filter(
-                            Aggregation.newAggregation(
-                                Aggregation.match(
-                                    Criteria.where("operationType")
-                                        .`in`(changeStreamOptionsConfig.operationType)
-                                ),
-                                Aggregation.project(changeStreamOptionsConfig.project)
-                            )
+            Flux.defer {
```

Comment:
Why is flux defer needed here? there is only a subscriber to this flux, so the _defer_ part to instantiate a new flux for each subscriber is useless here, right?

Reply:
Defer is needed when using retryWhen since it works by re-subscribing to the result of upstream methods